aten/src/ATen/native/vulkan/VulkanConvolution.cpp

```diff
@@ -66,14 +66,14 @@ ContextConv2D create(
   const auto stride_expanded = expand_param_if_needed(stride, "stride", 2);
   const auto dilation_expanded =
       expand_param_if_needed(dilation, "dilation", 2);
-  const Tensor weight_nchw = weight.contiguous();
+  Tensor weight_nchw = weight.contiguous();
+  auto ws = weight_nchw.sizes();
+  std::array<int64_t, 4> weight_sizes = {ws[0], ws[1], ws[2], ws[3]};
   return ContextConv2D{
-      at::native::vulkan_convolution_prepack_weights(weight),
+      groups == 1 ? at::native::vulkan_convolution_prepack_weights(weight_nchw)
+                  : std::move(weight_nchw.vulkan()),
       bias.has_value() ? c10::make_optional((*bias).vulkan()) : c10::nullopt,
-      {weight_nchw.sizes()[0],
-       weight_nchw.sizes()[1],
-       weight_nchw.sizes()[2],
-       weight_nchw.sizes()[3]},
+      weight_sizes,
```

Comment:
You could still pass this inline with the aggregate initializer list as you were doing, after having factored out weight_nchw.sizes().  Not that it should matter if the compiler is optimizing out the copy, as it should, but generally speaking using temporaries is a safer bet when possible.

This is merely a suggestion.  What you have is perfectly fine.